tensorflow_probability/examples/jupyter_notebooks/Probabilistic_Matrix_Factorization_TFP.ipynb

```diff
@@ -0,0 +1,926 @@
AgCbB\nAQAAaFpZdgFD2dzbSZJs7177vI82h7K+ujZo+111radr/5x0eze2jm5n6PHqakz1TGGuHTW+Q8+1\nsfXP0KZ+bBu6/THtu8dx3vpnyO0d++diS9d6Fq1/1O9VU++fsR37++6f2y+fpJoZZxwAAIAmwQEA\nAGgSHAAAgCbBAQAAaBIcAACAJsEBAABoEhwAAIAmwQEAAGgSHAAAgCbBAQAAaBIcAACAppVlFzCU\n9dW1JMnG1rXPT2Jzb+dYNQxlbPV01bX+g7Z3+2nntHTp/6lsU18Wbe8yx3fq/T+2fb2roY9tUz92\nDu2k/dPn5+5pGHJ/P4u/B0xtfIc0tmNJq567e3gPZxwAAIAmwQEAAGgSHAAAgCbBAQAAaBIcAACA\nJsEBAABoEhwAAIAmwQEAAGgSHAAAgCbBAQAAaBIcAACAppU+GimlPCDJv0zyj5J8UZI/S/JbSX6g\n1rp9YN17J3lBkicnuTnJXUnemeTFtdYP9lHP9drc2+m0/vrq2qDts1xTHt+utXc19LYuqn9j6/Bl\nXetZRv19Om/bO3T7XY1pX0+GP1adtP3t3eO97xSM7XNibPtKV2PrT78Xtp34jEMp5YFJtpN8e5J/\nO3/8X5L8/SS/VUp52L51LyR5a5IXJfnNJP8syf+c5LFJ3ltK+ZKT1gMAAPSvjzMOL09yU5JvrrW+\n+eqLpZT3J/mlzM4uPGn+8pOTfF2SV9Zan7dv3Xck+UCSVyZ5Qg81AQAAPerjGoe9JG9M8pYDr//7\nJFeS/J19rz1t/vjj+1ec/znTe5J8Qynl/j3UBAAA9OjEZxxqrS89ZNENSS5kdg3DVbcmuaPWurtg\n/fcleVSSWzK75gEAABiJC1euXBmk4VLKi5L8QJJn11p/rJRyQ2Yh4r211kcuWP9ZSV6V5Dtrra+5\n3ve58847F27ApUuXjlc4AACcMRcvXlz4+o033njhetsY5HaspZSvz+wuS7+d5F/PX75h/vipQ37s\nkwfWAwAARqKX27HuV0p5WpLXJPlIksfXWj/T93tcjxfc+ookycbW8695vt95u+3W2G4ZeVLbu29M\nktxy01N6aW9Mtzgcem52tZzbsfa37w5t6v3fVR/be9T4Dm1s/dnVmI5Vi9rv+9g8JmP7PWAZt2O1\n737O2OZDy935wInb6PWMQynlxUlen+R3knx1rfWP9y2+eq3DfQ758fseWA8AABiJ3oJDKeVVSV6W\n5JeTPKbW+rH9y2utn0jy8cxu3brIzfNHFycAAMDI9BIc5mcanpXkdUmeUGs97DqG9yS5qZTyoAXL\nHp3k05l9mRwAADAifXxz9Nck+f7MvsfhO2qtnz1i9dfOH59zoI3HJHl4kjfNz0wAAAAj0sfF0T80\nf/y1JE8opSxa5+211k/VWn+llPLmJM8updwvs+9ruDnJbUl2k3xfD/Vct7FfYNZ3+0M7bxeIDmls\nY9tVH3N/e/fwZVPvn66mvm8NPb5jOzaPzdDz52D7G1tHv+/Y+nNMN64YW9903XeHNrZjQ1fL/uy6\n/fLJ2+gjONwyf3z1Ees8OLO7LCXJU5J8b5KnJvnWJJeTvC3JC2utH+2hHgAAoGd9fHP0dX9pxHz9\nz2R2EfXLTvreAADA6RjkC+AAAICzRXAAAACaBAcAAKBJcAAAAJoEBwAAoElwAAAAmgQHAACgSXAA\nAACaBAcAAKDpxN8cPVabeztJku3da5/vt7661qnNrut31bX9RdvUp7HVc56MbWzHNhfG1j9d6xlb\nfw5t0fZubB2+bOqGHt+h59vQxrZ/jaXtMeq6747t2D/0eI3t2Hwa89MZBwAAoElwAAAAmgQHAACg\nSXAAAACaBAcAAKBJcAAAAJoEBwAAoElwAAAAmgQHAACgSXAAAACaBAcAAKBpZdkFTMnm3k6n9ddX\n1wZtvyv19GtM9XetZWh91LOx1V9bUzf0XBtb+11p/2hjG6+xHcuHNPS2LmNf3949fNmYPheP0/7Y\nPm/67s+7T1LMnDMOAABAk+AAAAA0CQ4AAECT4AAAADQJDgAAQJPgAAAANAkOAABAk+AAAAA0CQ4A\nAECT4AAAADQJDgAAQNPKsgtYps29nU7rr6+uDVTJ6bQ/9PZ2bf+k9WxsLX79tOrpqkt/nnZf9q2P\nvtze7a+tqe+7XQ297w5tbPWMrT/Htv+Obf8asv/Htq1djW3f4mh974u3Xz5JNTPOOAAAAE2CAwAA\n0CQ4AAAATYIDAADQJDgAAABNggMAANAkOAAAAE2CAwAA0CQ4AAAATYIDAADQJDgAAABNK8suYCjr\nq2tJko2ta5+fxObezonbOErXGrvWM7b2hzZ0PUPOhynXngw/16ZubP3P0RwLj3awf7Z3F79+3Hrs\nL4eb+rFzbH1vrrU54wAAADQJDgAAQJPgAAAANAkOAABAk+AAAAA0CQ4AAECT4AAAADQJDgAAQJPg\nAAAANAkOAABAk+AAAAA0rSy7gClZX11bdgmnquv2bu7tDFTJzNj6f0z907WWoWvvuv6ieja2Dl+2\njHo4vkX9v717+LKhnbfxPe1j81H7bh/t920Zc3BZ+ujLvseXaXHGAQAAaBIcAACAJsEBAABoEhwA\nAIAmwQEAAGgSHAAAgCbBAQAAaBIcAACAJsEBAABoEhwAAIAmwQEAAGhaWXYBy7S5tzNo++ura4O2\n39XQ2zu0g/Vv7y5+/aqu/T+m+dC1ljHVnvRTf2t8hzS2/hza0OO7qP2NrcOX9dF+n8Y2H8beP33v\nu0PXP+SxeWjLmPvLPDaPzdiOPafx2eKMAwAA0CQ4AAAATYIDAADQJDgAAABNggMAANAkOAAAAE2C\nAwAA0CQ4AAAATYIDAADQJDgAAABNggMAANC0suwChrK5t5Mk2d699vlJrK+uHauGodofup6xObi9\nG1uLX79qbNvbpZ6uYzu0oefyIq3xn
```

Comment:
tf.zeros([D, N], dtype=...) and tf.ones([D, N], ...). Here and elsewhere. I think also the # Distributions comments can be removed.